File: src/Entity/EventCategory.php
<?php

namespace App\Entity;

use App\Repository\EventCategoryRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: EventCategoryRepository::class)]
class EventCategory
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "You have to assign a discription")]
    #[Assert\Length(
        min: 7,
        max: 100,
        minMessage: "Description must be at least {{ min }} characters long",
        maxMessage: "Description cannot be longer than {{ max }} characters",
    )]
    private ?string $description = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "You have to assing a name")]
    private ?string $name = null;

    #[ORM\OneToMany(mappedBy: 'eventCategory', targetEntity: Event::class)]
    private Collection $events;

    public function __construct()
    {
        $this->events = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): self
    {
        $this->name = $name;

        return $this;
    }

    /**
     * @return Collection<int, Event>
     */
    public function getEvents(): Collection
    {
        return $this->events;
    }

    public function addEvent(Event $event): self
    {
        if (!$this->events->contains($event)) {
            $this->events->add($event);
            $event->setEventCategory($this);
        }

        return $this;
    }

    public function removeEvent(Event $event): self
    {
        if ($this->events->removeElement($event)) {
            // set the owning side to null (unless already changed)
            if ($event->getEventCategory() === $this) {
                $event->setEventCategory(null);
            }
        }

        return $this;
    }
    public function getEventCount(): int
    {
        return $this->events->count();
    }
}

File: src/com/job4u/entities/EventCategory.java
package com.job4u.entities;

public class EventCategory {

    private int id;
    private String description;
    private String name;

    public EventCategory() {
    }

    public EventCategory(int id, String description, String name) {
        this.id = id;
        this.description = description;
        this.name = name;
    }

    public EventCategory(String description, String name) {
        this.description = description;
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }


}